File: Java-Production-Line/src/Factory.java
/*
 * DEREN BOZER
 * COSC-525 MW 1:00PM
 * WINTER 2022
 * PROGRAMMING ASSIGNMENT: PRODUCTION LINE
 */

public class Factory {
    public static AnimationCube[] cubeArray = new AnimationCube[26];

    public static void main(String[] args) {

        // CREATES THE ANIMATION WINDOW
        new AnimationFrame();

        // CREATE THE THREE BUFFER CONVEYOR BELTS
        BoundedBuffer bufferAtoB = new BoundedBuffer("Belt A-B");
        BoundedBuffer bufferBtoC = new BoundedBuffer("Belt B-C");
        BoundedBuffer bufferCtoD = new BoundedBuffer("Belt C-D");
        BoundedBuffer bufferDtoFinal = new BoundedBuffer("Exit Belt");

        // CREATE THE WORKERS
        Worker workerA = new Worker("Worker A", true, bufferAtoB);
        Worker workerB = new Worker("Worker B", false, bufferAtoB, bufferBtoC);
        Worker workerC = new Worker("Worker C", false, bufferBtoC, bufferCtoD);
        Worker workerD = new Worker("Worker D", false, bufferCtoD, bufferDtoFinal, true);

        // START THE WORKER THREADS
        workerA.start();
        workerB.start();
        workerC.start();
        workerD.start();

        // JOINS THE THREADS WHEN THEY ARE FINISHED
        try {
            workerA.join();
            workerB.join();
            workerC.join();
            workerD.join();
        } catch (Exception e) { }

        // FINAL OUTPUT
        System.out.println("--------------------------------------------");
        System.out.println("Widget Processing complete. Final results:");
        System.out.println();
        bufferDtoFinal.print();
        System.out.println("--------------------------------------------");
        System.out.println();
        System.out.println("Program closing.");
    }
}

File: c#-Assembly-Line/AssemblyLine/Factory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace AssemblyLine
{
    public class Factory
    {
        public static void Main()
        {
            Console.WriteLine("Hello World");

            BoundedBuffer bufferAtoB = new BoundedBuffer();
            BoundedBuffer bufferBtoC = new BoundedBuffer();
            BoundedBuffer bufferCtoD = new BoundedBuffer();
            BoundedBuffer bufferDtoFinal = new BoundedBuffer();

            Worker objA = new Worker("WorkerA",true,bufferAtoB);
            Worker objB = new Worker("WorkerB", false, bufferAtoB, bufferBtoC);
            Worker objC = new Worker("WorkerC", false, bufferBtoC, bufferCtoD);
            Worker objD = new Worker("WorkerD", false, bufferCtoD, bufferDtoFinal, true);

            Thread threadA = new Thread(new ThreadStart(objA.run));
            Thread threadB = new Thread(new ThreadStart(objB.run));
            Thread threadC = new Thread(new ThreadStart(objC.run));
            Thread threadD = new Thread(new ThreadStart(objD.run));

            threadA.Start();
            threadB.Start();
            threadC.Start();
            threadD.Start();

            threadA.Join();
            threadB.Join();
            threadC.Join();
            threadD.Join();

            Console.WriteLine("--------------------");
            Console.WriteLine("Widget Processing Complete. Final results:");
            Console.WriteLine();
            bufferDtoFinal.print();
            Console.WriteLine("--------------------");
            Console.WriteLine();
            Console.WriteLine("Program Closing");
        }
    }
}
